client/my-sites/post-selector/selector.jsx

```diff
@@ -97,136 +109,202 @@ const PostSelectorPosts = React.createClass( {
-		const postType = get( this.props.postTypes, [ item.type, 'labels', 'singular_name' ], '' );
+	getItem( index ) {
+		if ( this.props.postsHierarchy ) {
+			return this.props.postsHierarchy[ index ];
+		}
+	},
 
-		const input = (
-			<input
-				id={ domId }
-				type={ inputType }
-				name="posts"
-				value={ itemId }
-				onChange={ this.props.onChange.bind( null, item ) }
-				checked={ checked }
-				className="post-selector__input" />
-		);
+	isCompact() {
+		if ( ! this.props.posts || this.state.searchTerm ) {
+			return false;
+		}
 
-		return (
-			<li key={ 'post-' + itemId } className="post-selector__list-item">
-				<label>
-					{ input }
-					<span className="post-selector__label">
-						{ decodeEntities( name ) }
-						<span className="post-selector__label-type">
-							{ decodeEntities( postType ) }
-						</span>
-					</span>
-				</label>
-				{ item.items ? this.renderHierarchy( item.items, true ) : null }
-			</li>
-		);
+		return this.props.posts.length < this.props.searchThreshold;
 	},
 
-	onSearch( event ) {
-		const newSearch = event.target.value;
-
-		if ( this.state.searchTerm && ! newSearch.length ) {
-			this.props.onSearch( '' );
+	isTypeLabelsVisible() {
+		if ( 'boolean' === typeof this.props.showTypeLabels ) {
+			return this.props.showTypeLabels;
 		}
 
-		if ( newSearch !== this.state.searchTerm ) {
-			analytics.ga.recordEvent( this.props.analyticsPrefix, 'Performed Post Search' );
-			this.setState( { searchTerm: event.target.value } );
-			this.debouncedSearch();
-		}
+		return 'any' === this.props.query.type;
 	},
 
-	renderHierarchy( items, isRecursive ) {
-		const listClass = isRecursive ? 'post-selector__nested-list' : 'post-selector__list';
+	isRowLoaded( { index } ) {
+		return this.props.lastPage || !! this.getItem( index );
+	},
 
-		return (
-			<ul className={ listClass }>
-				{ items.map( this.renderItem, this ) }
-				{
-					this.props.loading && ! isRecursive ?
-					this.renderPlaceholderItem() :
-					null
-				}
-			</ul>
-		);
+	getItemHeight( item ) {
+		let height = ITEM_HEIGHT;
+
+		if ( item && item.items ) {
+			height += item.items.reduce( ( memo, nestedItem ) => {
+				return memo + this.getItemHeight( nestedItem );
+			}, 0 );
+		}
+
+		return height;
 	},
 
-	renderPlaceholderItem() {
-		const inputType = this.props.multiple ? 'checkbox' : 'radio';
+	getRowHeight( { index } ) {
+		return this.getItemHeight( this.getItem( index ) );
+	},
 
-		return (
-			<li>
-				<input className="post-selector__input" type={ inputType } name="posts" disabled={ true } />
-				<label><span className="placeholder-text">Loading list of options...</span></label>
-			</li>
-		);
+	getResultsWidth() {
+		const { selectorRef } = this.state;
+		if ( selectorRef ) {
+			return selectorRef.clientWidth;
+		}
+
+		return 0;
 	},
 
-	renderPlaceholder() {
-		return ( <ul>{ this.renderPlaceholderItem() }</ul> );
+	getRowCount() {
```

Comment:
>So this row count excludes child `items` correct?  But this isn't an issue since the `getItemHeight` factors them in to the height correct?

Correct on both accounts.